manifests/charts/istiod-remote/templates/remote-istiod-service.yaml

```diff
@@ -1,13 +1,12 @@
+# This file is only used for remote `istiod` installs.
 {{- if .Values.global.remotePilotAddress }}
 apiVersion: v1
 kind: Service
 metadata:
-  {{- if .Values.enabled }}
-  # when local istiod is enabled, we can't use istiod service name to reach the remote control plane
-  name: istiod{{- if .Values.revision }}-{{ .Values.revision}}{{- end }}-remote
-  {{- else }}
-  # when local istiod isn't enabled, we can use istiod service name to reach the remote control plane
+  {{- if .Values.global.externalIstiod }}
```

Comment:
Would this ever be false for istiod-remote? Might be similar to the discussion here about pilot.enabled: https://github.com/istio/istio/pull/53188#discussion_r1765309971.

Reply:
Trying to see how much we can consolidate / eliminate values if possible.

Reply:
For reference: https://github.com/istio/istio.io/pull/15459#discussion_r1688485363

Reply:
It is not always set, but you cannot set both `externalIstiod` and `remotePilotAddress` at the same time (or, rather, today, if you do that, things will break).

This is not documented anywhere but the integration tests.

Reply:
Ok - so with your change, if `externalIstiod` and `remotePilotAddress` are both set, it will use the `istiod{{- if .Values.revision }}-{{ .Values.revision}}{{- end }}` it won't use the remote service name anymore? 

If it's not expected to set both, can we just get rid of `externalIstiod` from the istiod-remote helm chart?

Reply:
> If it's not expected to set both, can we just get rid of externalIstiod from the istiod-remote helm chart?

Yes probably. I wanna do that separately tho.

Reply:
https://github.com/istio/istio/issues/53288

for properly separating `istio-discovery` and `istiod-remote` and removing unneeded values from one or the other.

Reply:
Actually, I would prefer to merge `istio-discovery` and `istio-remote` into a single chart with fewer, more logical flags. But either option is out of scope here.